### ISSUE TITLE
Logo excepted from 1.4.3 Contrast (Minimum), but does it fail as a link for 1.4.11 Non-text Contrast?

### ISSUE THREAD

**Issue by jake-abma** [w3c/wcag#1739]
Logo excepted from 1.4.3 Contrast (Minimum), but does it fail as a link for 1.4.11 Non-text Contrast?

Logos are excepted from 1.4.3, that's clear and logic.

The same logo used as a link on a background without having enough contrast, is that a fail?
This is the core question as they might serve different needs.

The reason is the two parts of 1.4.11:

> **User Interface Components**
> Visual information required to identify user interface components and states, except for inactive components or where the appearance of the component is determined by the user agent and not modified by the author;
> 
> **Graphical Objects**
> Parts of graphics required to understand the content, except when a particular presentation of graphics is essential to the information being conveyed.

First part "User Interface Components":

The logo fails, the visual information is not enough, and there is NO exception.

Second part "Graphical Objects":

The logo passes, because it's excepted.

Also in the Understanding doc it says:

> Required for Understanding
> 
> The graphic is part of a logo or brand name (which is considered "essential" to its presentation).

"essential to its presentation" might be for from a brand perspective, not from a link perspective.

Further on it again says:

> Essential Exception
> Graphical objects do not have to meet the contrast requirements when "a particular presentation of graphics is essential to the information being conveyed". The Essential exception is intended to apply when there is no way of presenting the graphic with sufficient contrast without undermining the meaning. For example:
> 
> Logotypes and flags: The brand logo of an organization or product is the representation of that organization and therefore exempt. Flags may not be identifiable if the colors are changed to have sufficient contrast.

Now that is true for the logo, but applied to the link now it becomes the problem not being recognized as a interactive element / User Interface Component.

The explanation for the exception is also only for the understanding of the graphical identification, not for serving as a link where the problem lies.

So focussing on "User Interface Components" and the need, it seems we mixed the two needs, and end up with possibly lots of unrecognizable links, needed to navigate a site properly, while this SC is trying to prevent them!

Anything we can do about it?

Example here:

https://dequeuniversity.com/demo/dream#

The demo page from Deque with the purple, clickable logo, in the top left corner with the text "Dream Destination"

**Comment by JAWS-test** [comment]
See https://github.com/w3c/wcag/issues/902 and https://github.com/w3c/wcag/issues/1275

**Comment by jake-abma** [comment]
good call @JAWS-test 

My suggestion would be to except logo's from a branding perspective BUT prevent them from being used as interactive elements if the contrast is not good.

So in this case you can fix the background or not used it and forbid the use.

**Comment by fstrr** [comment]
> My suggestion would be to except logo's from a branding perspective BUT prevent them from being used as interactive elements if the contrast is not good.

This would mean that the common pattern of using a linked logo in a site header to link to a site's homepage wouldn't conform. I can't see how this would ever work—unless I'm missing something, this would require every company to have a WCAG-conforming logo now and, assuming the Silver contrast algorithm changes things, re-brand based on that.

**Comment by jake-abma** [comment]
@fstrr no it is not, we just say ALSO provide a proper contrasting background (which was the last sentence)

> So in this case you can fix the background or not use it

**Comment by alastc** [comment]
I'm not sure it is a big deal in terms of what would be required; assuming we agree that in this scenario logos do require contrast. 

Any part of the logo would need to contrast with some/all of it's background. For example, the demo page from Deque has 4.3:1 contrast, at least in some areas. 

The requirements for UICs in 1.4.11 is pretty basic for links/buttons. Where a logo is equivalent to the text in a link but you don't have to read it.

**Comment by gjccopinga** [comment]
After a discussion on this topic in the netherlands maybe I have a new approach to this. This is how I read the succescriterion and the explanation in the understanding document:

If there is visual information that identifies that 'something' is interactive then that visual information must have a contrast of 3.0:1 or more. So, If I have a text and it is underlined to identify it as a link, then that line must have enough contrast. Or, if I use a checkbox then I can recognise it is an interactive checkbox of the square border has enough contrast.

If I have a logo on my homepage which is not interactive and I have another page where it is interactive (link to the homepage) and they both look exactly the same, then there is no visual informatie that tells me that the logo on the other page is interactive. If there is no visual information to identify the logo as a component of the user interface there is no contrast requirement.
If the contrast is too low and you enhance the contrast of the logo it will only tell me the information of what it is and if it would be a link what the link purpose would probably be (not always, because sometimes is does not link to the homepage of the website, but to the homepage of a subsite with a special purpose). Enhancing the contrast does not make it more recognisable or identifiable als an interactive component. I myself would still have to hover my mouse over the logo to see if it really is interactive. There is no visuel information that tells me that it is interactive. Enhancing the contrast does not help with that.

So, as I read it, the logo as a link is still an exception, because we can only apply the 'graphical object' part of 1.4.11. Because there is no visual information that identifies the logo as an interactive component of the user interface.
